mantidproject/mantid
# Fix error when fitting a single domain from the Fit Script Generator

**Description of work.**
This PR fixes a bug with a python fit script generated by the Fit Script Generator. The index error would only appear when fitting a function to a single domain.

**To test:**
1. `Interfaces`->`General`->`Fit Script Generator`
2. Load the data files below into the workbench
3. Click `Add domains`  to add a single domain into the FSG. Then close the Add Domains dialog box
4. Change the End X to be 15.0
6. Look at the top right and make sure `All Domains` is selected.
7. Right click the area under the combo box and add a `GausOsc` function.
8. Change `Frequency` to have value `10.0`
9. Now click `Generate Script to Clipboard`
10. Open a new python script window on Workbench, and make sure it is empty. Right click and paste the generated code.
11. Run this python script and wait. You should get a plot of a single fit and its difference

**Data files**
[MUSR62260; Group; bkwd; Asymmetry; MA.zip](https://github.com/mantidproject/mantid/files/8473501/MUSR62260.Group.bkwd.Asymmetry.MA.zip)

Fixes #33754





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
